app/assets/javascripts/table-sort.es6

```diff
@@ -24,9 +25,33 @@ $(document).ready(function() {
             return this.parentNode;
           });
           inverse = !inverse;
-
+          
```

Comment:
The way this is coded is far more complicated than it needs to be. Basically, the code above this comment is creating some output and putting it into the page, and the code below is gathering that data again from the page and parsing it to work out the same logic.

For example: The parent `if` statements between lines 31 and 47 are basically saying "is there a tag with the class set so it sorts like ...." - and not just in a restricted part of the page - but the whole page. It then uses the result of the searches to make decisions on what the screen reader message should be.

When: 

- The code already knows which html element is being used to sort the table. It's defined above as `th`.
- The code already knows which direction the sort is occurring (ascending/descending). It's defined by `inverse` (and see my other comment about using that to define `sortDirection`).
- The first part of the screen reader message will be the text content of element `th`. It should always be `th.text()`.

So: 

What's needed is to work out the data type: "numeric", "alphabetic", "description". At the moment that's done by testing for one of three HTML classes. It would be much easier if a `th` element being numerical was assigned using a `data-` attribute rather than a class. So:
```html
<th data-sort-type="numeric">1</th>
<th data-sort-type="alphabetic">2</th>
<th data-sort-type="description">3</th>
```
With that in place, the sort type will be returned by `th.dataset.sort-type`

We now have the starting text (`th.text()`), the sort direction (`sortDirection`) and the sort type (`th.dataset.sort-type`). We next need to workout the ending text based on sort direction and type. We can use a matrix for that:
```javascript
endText = {
  asc: {
    numberic: 'from big to small',
    alphabetic: 'in reverse order',
    description: 'in <em>ascending</em> order'
  },
  desc: {
    numberic: 'from small to big',
    alphabetic: 'in alphabetical order',
    description: 'in <em>descending</em> order'
  }
}
```
With all that in place, we can define the construction of the screen reader message once:
```javascript
$("#screen-reader-messages").html("Sorted by " + th.text() + " " + endText[sortDirection][th.dataset.sort-type]);
```

Reply:
Oh - and this line will need to be moved to after the screen reader message has been set:
```javascript
inverse = !inverse;
```

Reply:
Changes made.  Except:
`th.dataset.sort-type` is throwing an error, so I am using `th.attr("data-sort-type")`.
`th.text()` contains the "click to sort by" message, so I'm using `th.find(".aria-sort-description").text()`
Thanks.